.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        node: [12.x, 14.x, 15.x]
+      fail-fast: false
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set node version to ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Versions
+        run: yarn versions
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      # - name: Check outdated dependencies
+      #   if: github.ref == 'refs/heads/main'
+      #   run: yarn outdated
+
+      - name: Build vite
+        run: yarn build-vite
+
+      - name: Build plugin-vue
+        run: yarn build-plugin-vue
+
+      - name: Lint
+        if: matrix.os == 'ubuntu-latest' && matrix.node == '14.x'
```

Comment:
Instead of using a condition, you can move this to a separate parallel job.

Something like this:
https://github.com/atom-community/atom-ide-template/blob/61142e31b017526585cb9b6cb3564b2123ebd855/.github/workflows/CI.yml#L34-L60